helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/MakerCheckerSteps.java

```diff
@@ -0,0 +1,135 @@
+
+  private final MakerCheckerHelper makerCheckerHelper;
+
+  private final UserHelper userHelper;
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  public MakerCheckerSteps(MessageHelper messageHelper, MakerCheckerHelper makerCheckerHelper, UserHelper userHelper) {
+    this.messageHelper = messageHelper;
+    this.makerCheckerHelper = makerCheckerHelper;
+    this.userHelper = userHelper;
+  }
+
+  @When("$agent agent sends an attachment $attachment")
+  public void sendAttachment(String agent, String attachment)
+      throws SymException, URISyntaxException {
+    URL url = this.getClass().getClassLoader().getResource(ATTACHMENTS_DIR + attachment);
+
+    File fileAttachment = new File(url.toURI());
+
+    List<SymAttachmentInfo> attachmentsList = new ArrayList<>();
+    SymAttachmentInfo symAttachmentInfo = new SymAttachmentInfo();
+    symAttachmentInfo.setId(UUID.randomUUID().toString());
+    symAttachmentInfo.setName(fileAttachment.getName());
+    attachmentsList.add(symAttachmentInfo);
+
+    SymMessage message = new SymMessage();
+    message.setMessageText("Check my attachment, please ?");
+    message.setAttachment(fileAttachment);
+    message.setAttachments(attachmentsList);
+
+    messageHelper.sendAgentMessage(agent, message);
+  }
+
+  @When("$agent agent approve the attachment")
+  public void approveAttachment(String agent)
+      throws MessagesException, HealthCheckFailedException, MalformedURLException {
+    SymUser userAgent = userHelper.getUser(agent.toUpperCase());
+    Optional<SymMessage> message = messageHelper.getLatestTicketMessage(userAgent, initialTime);
+
+    if (!message.isPresent()) {
+      throw new TicketRoomNotFoundException("Message not found");
+    }
+
+    String url = getValueFromParam(message.get().getEntityData(), APPROVE_URL);
+
+    makerCheckerHelper.actionAttachment(url, userAgent.getId());
+  }
+
+  @When("$agent agent deny the attachment")
+  public void denyAttachment(String agent)
+      throws MessagesException, HealthCheckFailedException, MalformedURLException {
+    SymUser userAgent = userHelper.getUser(agent.toUpperCase());
+    Optional<SymMessage> message = messageHelper.getLatestTicketMessage(userAgent, initialTime);
+
+    if (!message.isPresent()) {
+      throw new TicketRoomNotFoundException("Message not found");
+    }
+
+    String url = getValueFromParam(message.get().getEntityData(), DENY_URL);
+
+    makerCheckerHelper.actionAttachment(url, userAgent.getId());
+  }
+
+  @Then("$agent can verify the attachment is $state")
+  public void verifyStateOfMakerChecker(String agent, String state)
+      throws MessagesException, HealthCheckFailedException {
+    SymUser userAgent = userHelper.getUser(agent.toUpperCase());
+    Optional<SymMessage> message = messageHelper.getLatestTicketMessage(userAgent, initialTime);
+
+    assertEquals(state.toUpperCase(), getValueFromParam(message.get().getEntityData(), STATE));
```

Comment:
Please validate the approved/denied message posted in the ticket room.

Reply:
Done.